File: fyc-api-rest/src/main/java/fr/esgi/fyc_api_rest/service/SportifService.java
package fr.esgi.fyc_api_rest.service;

import fr.esgi.fyc_api_rest.business.Sportif;

import java.util.List;

public interface SportifService {
    Sportif create(Sportif sportif);
    List<Sportif> findAll();
    Sportif findById(Long id);
    Sportif update(Long id, Sportif sportif);
    void delete(Long id);
}

File: fyc-rest-api-csharp-test/CometeAPI/Service/SportifService.cs
using FycRestApi.Business;
using FycRestApi.Repository;
using FycRestApi.Service.Interface;

namespace FycRestApi.Service;

public class SportifService : ISportifService
{
    private readonly SportifRepository _sportifRepository;
    public SportifService(SportifRepository sportifRepository)
    {
        _sportifRepository = sportifRepository;
    }

    public async Task Delete(int id)
    {
        await _sportifRepository.Delete(id);
    }

    public Task<List<Sportif>> FindAll()
    {
        return _sportifRepository.FindAll();
    }

    public Task<Sportif> FindById(int id)
    {
        return _sportifRepository.FindById(id);
    }

    public async Task<Sportif> Save(Sportif sportif)
    {
        return await _sportifRepository.Save(sportif);
    }

    public Task<Sportif> Update(int id, Sportif sportif)
    {
        return _sportifRepository.Update(id, sportif);
    }
}